docs/DeveloperGuide.md

```diff
@@ -315,16 +315,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  When a new version is released, it should be possible to update the most recent version if the user has installed the previous version.
+5.  Installing a new update shall not in any way, modify or erase existing data and value from the previous version, and the new update should be compatible with the data produced earlier within the system.
+6.  Should be able to store notes in English language, and provisions shall be made to support all languages.
+7.  The system should be able to handle notes with large capacity without any performance liability so that users can keep track of each activity without any issues.
+8.  Whenever a change is made to the application, the progress is instantly saved to the system.
+9.  The system should be able to reply to the prompt or command from the user within 3 seconds.
+10.  The system should cater to mentor professor such that mentor professor can access the application without any difficulty.
+11.  Should ensure personal data privacy and security of data access. 
+12. Software testing will require the use of automated testing. The test will be deleted after successful implementation of the software system.
 
-*{More to be added}*
 
 ### Glossary
```

Comment:
For glossary, I think we might not want to include 
* Automated Testing -- seems to be almost a NFR?
* Help Page -- exactly the same meaning as common meaning
* Trie -- same meaning as is generally used, and maybe not necessary to all stakeholders to know
* Reminder -- I'm on the fence about this one, but mainly... are we even implementing reminders? Anyway, I think the meaning is quite intuitive...
* Ungrouped -- the definition suggests that "Ungrouped" is a group that we make to hold all people without a group, are we really doing that + does it actually need to be in our glossary?

other than that good job in coming up with these, I hadn't thought of a lot of them!

Reply:
I am still a bit doubtful about automated testing, we might want to discuss this at the meeting, I do think we can include this though.
Help page: it is the same haha :p will not include this.
trie: I think it is good to know about what we use for in the trie, and that is the tags and commands in this case, which is why I include this~ as for the stakeholders, I do think not everyone need to know haha. Regardless, I'll still put this in for now but we might want to discuss this.
Reminder: yeah, I also thought about whether or not we want to implement this :) for now I'll include it, but I try to make the meaning cater to our target audience (mentor prof)
ungrouped: I think it's best to include this to the glossary, so that we know people who weren't in any groups and hence, these people will be in "ungrouped" group haha. 

Thanks for the feedback! I'll amend a few things according to your suggestions :)